### PR TITLE
fix: test case multi-namespace_support

### PR DESCRIPTION
**Pull request by lihongyan1**
Fix issue #311 

Prometheus pod need more resources and has complicated configuration, replace it with prometheus example app pod

### CODE REVIEW

File: test/e2e/monitoring_stack_controller_test.go [hunk 1]
```diff
@@ -973,6 +973,7 @@ func newService(t *testing.T, name, namespace string, labels, selector map[strin
 }
 
 func newPrometheusPod(t *testing.T, name, ns string) *corev1.Pod {
```
Comment (sthaha) [inline test/e2e/monitoring_stack_controller_test.go:975]: could you please rename this to `newPrometheusExamplePod` now that the definition has changed?

File: test/e2e/monitoring_stack_controller_test.go [hunk 2]
```diff
@@ -983,16 +984,27 @@ func newPrometheusPod(t *testing.T, name, ns string) *corev1.Pod {
...
-				"app.kubernetes.io/version": "2.39.1",
+				"app.kubernetes.io/version": "multiarch-v0.1.0",
 			},
 		},
 		Spec: corev1.PodSpec{
 			Containers: []corev1.Container{{
-				Name:  "prometheus",
-				Image: "quay.io/prometheus/prometheus:v2.39.1",
+				Name:  "prometheus-example-app",
+				Image: "quay.io/openshifttest/prometheus-example-app@sha256:382dc349f82d730b834515e402b48a9c7e2965d0efbc42388bd254f424f6193e",
```
Comment (sthaha) [inline test/e2e/monitoring_stack_controller_test.go:993]: I wonder if we should instead rely on what upstream Prometheus Operator is using ?

File: test/e2e/monitoring_stack_controller_test.go [hunk 3]
```diff
@@ -983,16 +984,27 @@ func newPrometheusPod(t *testing.T, name, ns string) *corev1.Pod {
...
+				SecurityContext: &corev1.SecurityContext{
+					AllowPrivilegeEscalation: &bF,
```
Comment (sthaha) [inline test/e2e/monitoring_stack_controller_test.go:995]: nit: consider `pointer.Bool` - https://pkg.go.dev/k8s.io/utils/pointer#Bool

Comment (sthaha) [inline test/e2e/monitoring_stack_controller_test.go:995]: nit: consider `pointer.Bool` - https://pkg.go.dev/k8s.io/utils/pointer#Bool

File: test/e2e/monitoring_stack_controller_test.go [hunk 4]
```diff
@@ -861,7 +861,7 @@ func namespaceSelectorTest(t *testing.T) {
 
 	promClient := framework.NewPrometheusClient("http://localhost:9090")
 	if pollErr := wait.Poll(5*time.Second, 5*time.Minute, func() (bool, error) {
-		query := `prometheus_build_info{namespace=~"test-ns-.*"}`
+		query := `version{namespace=~"test-ns-.*"}`
```
Comment (sthaha) [inline test/e2e/monitoring_stack_controller_test.go:864]: is there a different metric that we can reply on that is specific to `prometheus_example` ?

**Review by sthaha — CHANGES_REQUESTED** [review]
Thank you for the patch @lihongyan1. The approach looks good to me. Some comments that needs addressing.

File: test/e2e/monitoring_stack_controller_test.go (hunk 2 above)
Comment (lihongyan1) [reply test/e2e/monitoring_stack_controller_test.go:993]: I would prefer to use this one, for I run the upstream test cases on OCP with all kinds of profiles share between QE team during ObO release, the image will be mirrored by default and the test cases can get the image and  run on disconnected or private OCP clusters, other wise, the case will failed on private and disconnected OCP cluster.

File: test/e2e/monitoring_stack_controller_test.go (hunk 4 above)
Comment (lihongyan1) [reply test/e2e/monitoring_stack_controller_test.go:865]: Yes, there are the following metrics, for detail you can refer [web site](https://github.com/brancz/prometheus-example-app)
```
version - of type gauge - containing the app version - as a constant metric value 1 and label version, representing this app version
http_requests_total - of type counter - representing the total numbere of incoming HTTP requests
http_request_duration_seconds - of type histogram, representing duration of all HTTP requests
http_request_duration_seconds_count- total count of all incoming HTTP requeests
http_request_duration_seconds_sum - total duration in seconds of all incoming HTTP requests
http_request_duration_seconds_bucket
```

Comment (lihongyan1) [reply test/e2e/monitoring_stack_controller_test.go]: I know what you mean, I changed the metric as `version{pod="prometheus-example-app",namespace=~"test-ns-.*"}`, thanks for the comments

File: test/e2e/monitoring_stack_controller_test.go (hunk 2 above)
Comment (jan--f) [reply test/e2e/monitoring_stack_controller_test.go:994]: @lihongyan1 would you mind leaving a comment that describes this and a link to the upstream that is mirrored? Just for future us when someone asks why we don't use the upstream version :)

Comment (lihongyan1) [reply test/e2e/monitoring_stack_controller_test.go:996]: Added comments

File: test/e2e/monitoring_stack_controller_test.go [hunk 5]
```diff
@@ -983,16 +985,29 @@ func newPrometheusPod(t *testing.T, name, ns string) *corev1.Pod {
...
+				"app.kubernetes.io/version": "multiarch-v0.4.1",
 			},
 		},
 		Spec: corev1.PodSpec{
 			Containers: []corev1.Container{{
-				Name:  "prometheus",
-				Image: "quay.io/prometheus/prometheus:v2.39.1",
+				Name: "prometheus-example-app",
+				//To suppose multi-arch, image built directly from https://github.com/brancz/prometheus-example-app
+				//To support OCP disconnected cluster and private cluster, put image under quay.io/openshifttest to be mirrored by default
```
Comment (sthaha) [inline test/e2e/monitoring_stack_controller_test.go:995]: How about .. 

```suggestion
				// The following image is a  rebuild  of  https://github.com/brancz/prometheus-example-app
				// with multi-arch enabled see: https://quay.io/repository/openshifttest/prometheus-example-app/manifest/sha256:382dc349f82d730b834515e402b48a9c7e2965d0efbc42388bd254f424f6193e 
				// In addition to the above, this image is also available in OCP disconnected cluster,
				// thus allowing the tests to be run in that environment.
```

**Review by sthaha — APPROVED** [review]
lgtm

**Pull request merged**